### PR TITLE
support pip install

### PR DESCRIPTION
**Pull request by yodada**
This PR contains a few tricks to allow
```
pip install git+https://github.com/cornell-brg/hb-pytorch.git
```

 - added a method `setup_emul` to `setup.py`, which sets a few ENV
 - if `setup.py` is called with `egg_info`, do not build torch. We use this pass to install dependencies like pyyaml.
 - if `setup.py` is called with `--record`, we trigger torch building.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -282,6 +292,30 @@ def report(*args):
...
+    os.environ["USE_QNNPACK"] = "0"
+    os.environ["USE_DISTRIBUTED"] = "0"
+    os.environ["USE_OPENMP"] = "0"
+    os.environ["ATEN_THREADING"] = "NATIVE"
+    os.environ["OMP_NUM_THREADS"] = "1"
+    os.environ["USE_HB_EMUL"] = "1"
+    if os.getenv("BSG_MANYCORE_DIR", "") != "":
+        del os.environ["BSG_MANYCORE_DIR"]
+    if os.getenv("HB_KERNEL_DIR", "") != "":
+        del os.environ["HB_KERNEL_DIR"]
```
Comment (vb000) [inline setup.py:317]: Couldn't understand this part... where do we set these variables?

Comment (yodada) [reply setup.py:317]: Precautions. Same as
https://github.com/cornell-brg/hb-pytorch/blob/413850923789b97b317edf43585d62e6a50d119e/setup_emul_build_env.sh#L29-L31

Comment (vb000) [reply setup.py:317]: Oh, sorry, this is emulation, we don't need them... my bad!

File: setup.py [hunk 2]
```diff
@@ -352,7 +386,9 @@ def check_file(f):
 ################################################################################
 
 # the list of runtime dependencies required by this built package
-install_requires = []
+install_requires = ['numpy', 'requests', 'six', 'sklearn', 'tqdm',
+                    'pytest', 'ninja', 'hypothesis', 'pyyaml',
+                    'typing']
```
Comment (vb000) [inline setup.py:391]: I wonder why pytorch didn't do this!

**Review by vb000 — APPROVED** [review]
Great!

**Pull request merged**